### PR TITLE
[FIX] simplified OPXLHelper::computeDeltaScores

### PR DESCRIPTION
**Pull request by enetz**
This should fix a complaint by the  AddressSanitizer in the nightly build

### CODE REVIEW

File: src/openms/include/OpenMS/ANALYSIS/XLMS/OPXLHelper.h [hunk 1]
```diff
@@ -218,7 +218,7 @@ namespace OpenMS
       static void addPercolatorFeatureList(ProteinIdentification& prot_id);
 
       /**
-       * @brief adds the delta score as a MetaValue
+       * @brief adds the delta score as a MetaValue, assumes the hits under each PeptideIdentification are ordered by score rank
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/ANALYSIS/XLMS/OPXLHelper.h]: can you add an OPENMS_PRECONDITION that checks if it is sorted?

Comment (enetz) [reply src/openms/include/OpenMS/ANALYSIS/XLMS/OPXLHelper.h]: is there a quick way to check if PeptideHits are sorted by score?

Comment (enetz) [reply src/openms/include/OpenMS/ANALYSIS/XLMS/OPXLHelper.h]: ah Ok, PeptideIdentification has a function to sort Hits by score. I guess I can just run that before calculating the delta score to make sure as well.

**Comment by jpfeuffer** [comment]
This is good! But I assume there is also something wrong during reading or writing mzIdentML with XL mods (see Windows machines).

File: src/openms/source/ANALYSIS/XLMS/OPXLHelper.cpp [hunk 2]
```diff
@@ -1188,27 +1188,14 @@ namespace OpenMS
...
-        PeptideHit& first = pep_id.getHits()[rank_indices[rank + 1]];
-        PeptideHit& second = pep_id.getHits()[rank_indices[rank]];
-
-        double delta_score = first.getScore()
-                             / second.getScore();
-
-        second.setMetaValue(Constants::UserParam::DELTA_SCORE, delta_score);
-      }
-      pep_id.getHits()[rank_indices.back()].setMetaValue(Constants::UserParam::DELTA_SCORE, 0.0);
+      phs[phs.size()-1].setMetaValue(Constants::UserParam::DELTA_SCORE, 0.0);
```
Comment (hroest) [inline src/openms/source/ANALYSIS/XLMS/OPXLHelper.cpp]: what if phs is empty?

**Comment by enetz** [comment]
@jpfeuffer the MzIdentML test seemed to fail because of the same function. It is also called when reading in mzid if it does not have a delta score.

File: src/openms/source/ANALYSIS/XLMS/OPXLHelper.cpp (hunk 2 above)
Comment (enetz) [reply src/openms/source/ANALYSIS/XLMS/OPXLHelper.cpp]: you are right, I will add a check

**Pull request merged**